### ISSUE TITLE
[Feature Request] Add new `inferOptions` util Typescript type

### ISSUE THREAD

**Issue by snelsi** [liaoliao666/react-query-kit#26]
## Context
`react-query-kit` already provides 3 util types:
```tsx
export type inferVariables<T> = ...;
export type inferData<T> = ...;
export type inferFnData<T> = ...;
```

## Describe your proposal
Would be great to also provide `inferOptions` type, that would return `QueryHookOptions` or `MutationHookOptions` of the given hook.

You can kinda achieve similar result manually by doing:
```tsx
type Options = QueryHookOptions<
  inferData<typeof usePost>,
  any,
  inferVariables<typeof usePost>,
  any
>;
```
or
```tsx
type Options = MutationHookOptions<
  inferData<typeof usePost>,
  any,
  inferVariables<typeof usePost>,
  any
>;
```
But it's not super convenient.

## Example of possible usage
```tsx
// Define custom hook
const usePost = createQuery<Response, Variables, Error>({
  primaryKey: '/posts',
})

// Infer it's options
type usePostOptions = inferOptions<typeof usePost>

// Extend your hook 
const useExtendedPost = (options: usePostOptions) => usePost({ ...options, myExtraParam: true }) 
```

**Comment by snelsi** [comment]
Possible implementation:
```tsx
export type inferOptions<T> = T extends QueryHook<infer TFnData, infer TVariables, infer TError>
? QueryHookOptions<TFnData, TError, TFnData, TVariables>
: T extends MutationHook<infer TData, infer TVariables, infer TError>
? MutationHookOptions<TData, TError, TVariables, any>
: never;
```

**Comment by liaoliao666** [comment]
Available in v1.5.2
